Assets/Scripts/InventoryAndItemsScripts/InventoryController.cs

```diff
@@ -0,0 +1,227 @@
+        coins = inventory.transform.GetChild(1).gameObject;
+        health = inventory.transform.GetChild(2).gameObject;
+        itemDesc = inventory.transform.GetChild(3).gameObject;
+
+        inventorySlots = new List<GameObject>();
+        itemsInInventory = new List<ItemData>();
+
+        foreach (Transform child in invPanel.transform)
+        {
+            inventorySlots.Add(child.gameObject);
+        }
+
+        selectedSlot = invPanel.transform.GetChild(0).gameObject.transform.GetChild(0).gameObject.GetComponent<UnityEngine.UI.Image>();
+    }
+
+    private void OnEnable()
+    {
+        InventoryAction.performed += OpenInventory;
+        InventoryAction.Enable();
+        SelectItem.performed += EquipItem;
+        SelectItem.Enable();
+
+        // Navigating inventory
+        NavLeft.performed += NavInvLeft;
+        NavRight.performed += NavInvRight;
+        NavUp.performed += NavInvUp;
+        NavDown.performed += NavInvDown;
+
+        NavLeft.Enable();
+        NavRight.Enable();
+        NavUp.Enable();
+        NavDown.Enable();
+
+    }
+
+    private void OnDisable()
+    {
+        InventoryAction.performed -= OpenInventory;
+        InventoryAction.Disable();
+        SelectItem.performed -= EquipItem;
+        SelectItem.Disable();
+
+        // Navigating inventory
+        NavLeft.performed -= NavInvLeft;
+        NavRight.performed -= NavInvRight;
+        NavUp.performed -= NavInvUp;
+        NavDown.performed -= NavInvDown;
+
+        NavLeft.Disable();
+        NavRight.Disable();
+        NavUp.Disable();
+        NavDown.Disable();
+    }
+
+    void OpenInventory(CallbackContext c)
+    {
+        if (inventory.activeSelf)
+        {
+            GameStateManager.Instance.SetState(new PlayingState());
+        }
+        else
+        {
+            // make sure something else isn't already pausing the game
+            if (GameStateManager.Instance.CurrentState is not PauseState) {
+                GameStateManager.Instance.SetState(new GameStateManagement.InventoryState(this));
+            }
+        }
+    }
+
+    void EquipItem(CallbackContext c)
+    {
+        if (inventory.activeSelf && itemsInInventory[inventoryIndex] != null)
+        {
+            pInv.EquipItem(itemsInInventory[inventoryIndex]);
+        }
+    }
+
+    void NavInvLeft(CallbackContext c)
+    {
+        if (inventory.activeSelf)
+        {
+            if (inventoryIndex > 0 && inventoryIndex % 6 != 0) 
+            {
+                inventoryIndex--;
+                selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+                selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+                Debug.Log(inventoryIndex);
+            }
+        }
+    }
+
+    void NavInvRight(CallbackContext c)
+    {
+        if (inventory.activeSelf)
+        {
+            if (inventoryIndex < inventorySlots.Count && inventoryIndex % 6 != 5)
```

Comment:
may want to factor out the 5s and 6s in this file, incase we ever want to change the layout of the inventory

Reply:
Completely agree